src/appleseed.shaders/include/appleseed/fractal/as_noise_helpers.h

```diff
@@ -59,4 +67,115 @@
     (1 - smoothstep(0.2, 0.75, filter_width)) * scale + p,              \
     filter_width)
 
+float noise_quadratic(vector control_p, float x)
+{
+    float tmp = control_p[0] - control_p[1];
+
+    return ((tmp - control_p[1] + control_p[2]) * x - tmp - tmp) *
+       x + control_p[0] + control_p[1];
+}
+
+float noise_derivative(vector control_p, float x)
+{
+    float tmp = control_p[0] - control_p[1];
+
+    return (tmp - control_p[1] + control_p[2]) * x - tmp;
+}
+
+vector noise_lookup(int y, int vx[3])
+{
+    float rng_table[65536] = { RNG_TABLE };
```

Comment:
@dictoon i wished. What i can tell you from the last week(s) is that in terms of (runtime) performance i don't notice anything abnormal, but the actual shader compilation times increase tremendously.